// assertio/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function assert(value, msg) {
    if (value !== true) {
        debugger;
        throw new Error(`Assertion failed${msg === undefined ? ': ' + msg : ''}`);
    }
}
exports.assert = assert;
function ensureArray(value) {
    if (!Array.isArray(value)) {
        throw createInvariant(value);
    }
    return value;
}
exports.ensureArray = ensureArray;
function ensureObject(value) {
    if (typeof value !== 'object' || value === null) {
        throw createInvariant(value);
    }
    return value;
}
exports.ensureObject = ensureObject;
function ensureString(value) {
    if (typeof value !== 'string') {
        throw createInvariant(value);
    }
    return value;
}
exports.ensureString = ensureString;
function ensureNumber(value) {
    if (typeof value !== 'number') {
        throw createInvariant(value);
    }
    return value;
}
exports.ensureNumber = ensureNumber;
function ensureBoolean(value) {
    if (typeof value !== 'boolean') {
        throw createInvariant(value);
    }
    return value;
}
exports.ensureBoolean = ensureBoolean;
function nonUndefined(value) {
    if (value === undefined) {
        throw createInvariant(value);
    }
    return value;
}
exports.nonUndefined = nonUndefined;
function nonNull(value) {
    if (value === null) {
        throw createInvariant(value);
    }
    return value;
}
exports.nonNull = nonNull;
function nonNullable(value) {
    if (value === null || value === undefined) {
        throw createInvariant(value);
    }
    return value;
}
exports.nonNullable = nonNullable;
function createInvariant(value) {
    debugger;
    return new Error(`Value should not to be ${JSON.stringify(value)}`);
}
